### ISSUE TITLE
Add `swiper-backward` command

### ISSUE THREAD

**Issue by vspinu** [abo-abo/swiper#1172]
It's would be great to have a counterpart of `isearch-backward` in swiper. 

The discussion of it, and a proposed solution, could be found in #412.  That solution is inherently fragile as most of it is a direct copy-paste of the swiper code. So even small changes to the internals of swiper are likely to break that solution.  I guess a small incision to the swiper code could fix this without much ado.

**Comment by manuel-uberti** [comment]
>  I guess a small incision to the swiper code could fix this without much ado.

Could you open a PR with it? I am interested to see this kind of command.

**Comment by braham-snyder** [comment]
@manuel-uberti  https://github.com/abo-abo/swiper/issues/412#issuecomment-324074376

**Comment by bbatsov** [comment]
I was wondering if a such command exists, now I know it doesn't. :-)

**Comment by abo-abo** [comment]
@vspinu Any chance on PR?

**Comment by vspinu** [comment]
No, not really, sorry. This part didn't quite stick with me, so I am back to usual isearch interface for the moment.

**Comment by andschwa** [comment]
Now that `swiper-isearch` exists it seems even more relevant to have `swiper-isearch-backward`.

**Comment by SreenivasVRao** [comment]
Hi guys, thanks for this feature! :) 

Can you tell me how I should remap C-r within swiper? I don't want it in the global keymap...

I tried 
`
(global-unset-key (kbd "C-r"))
 (define-key ivy-minibuffer-map (kbd "C-r") 'swiper-isearch-backward)`

didn't work...

**Comment by abo-abo** [comment]
@SreenivasVRao You should use `swiper-map`, which has priority over `ivy-minibuffer-map`.

**Comment by SreenivasVRao** [comment]
That caused some other problems. I looked at the key map and saw that C-s was bound to `'swiper-C-s`

So I went ahead and wrote a `'swiper-C-r` - Think this would be useful, but it assumes that people want C-s and C-r bound within swiper - which I know is not always the case.

`
(defun swiper-C-r (&optional arg)
  "Move cursor vertically down ARG candidates.
If the input is empty, select the previous history element instead."
  (interactive "p")
  (if (string= ivy-text "")
      (ivy-next-history-element 1)
    (ivy-previous-line arg)))
`
